follow fn calls in find_triton_kernels

dolpm opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #171218
* __->__ #171202

Summary:
follow-up to https://github.com/pytorch/pytorch/pull/171118, handling the todo about tracing through function calls

Test Plan:
python -m pytest test/dynamo/test_aot_autograd_cache.py::AOTAutogradCacheTests -k "test_triton_op" -v

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo

dolpm committed:
follow fn calls in find_triton_kernels

Summary:
follow-up to https://github.com/pytorch/pytorch/pull/171118, handling the todo about tracing through function calls

Test Plan:
python -m pytest test/dynamo/test_aot_autograd_cache.py::AOTAutogradCacheTests -k "test_triton_op" -v

[ghstack-poisoned]

dolpm committed:
Update on "follow fn calls in find_triton_kernels"


Summary:
follow-up to https://github.com/pytorch/pytorch/pull/171118, handling the todo about tracing through function calls

Test Plan:
python -m pytest test/dynamo/test_aot_autograd_cache.py::AOTAutogradCacheTests -k "test_triton_op" -v

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

zou3519 commented on torch/_library/triton.py:
```diff
@@ -31,18 +31,41 @@ def get_inner_triton_kernels(fn: Callable[..., Any]) -> list[object]:
 
-    def find_triton_kernels(fn: Callable[..., Any]) -> list[object]:
+    # prevent infinite recursion
+    MAX_RECURSION_DEPTH = 5
```
(we  shouldn't delete the comment about being best-effort if so)

dolpm replied:
there are probably other edge cases, anyway. this whole thing is best effort lol... we should probably put a comment at the top that states this

dolpm committed:
Update on "follow fn calls in find_triton_kernels"


Summary:
follow-up to https://github.com/pytorch/pytorch/pull/171118, handling the todo about tracing through function calls

Test Plan:
python -m pytest test/dynamo/test_aot_autograd_cache.py::AOTAutogradCacheTests -k "test_triton_op" -v

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

dolpm commented:
@pytorchmergebot merge

dolpm commented:
@pytorchmergebot merge
